### PR TITLE
[TT-5852]added sub request id to request

### PR DESCRIPTION
**Pull request by kofoworola**
## Description

This PR adds a header `X-Tyk-Parent-Request-Id` to subgraph upstream requests, this header value is the value of the initiating requests parent_id, this is to be used in pump to distinguish between subgraph requests, as well as tie them in with their corresponding supergraph

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4313/merge
Commit:  
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3124420625)

**Review by buraksezer — APPROVED** [review]

**Review by buraksezer — APPROVED** [review]

File: gateway/reverse_proxy.go [hunk 1]
```diff
@@ -1033,10 +1033,26 @@ func (p *ReverseProxy) handoverRequestToGraphQLExecutionEngine(roundTripper *Tyk
...
+		execOptions := []graphql.ExecutionOptionsV2{
 			graphql.WithBeforeFetchHook(p.TykAPISpec.GraphQLExecutor.HooksV2.BeforeFetchHook),
 			graphql.WithAfterFetchHook(p.TykAPISpec.GraphQLExecutor.HooksV2.AfterFetchHook),
-		)
+		}
+
+		// if this context vars are enabled and this is a supergraph, inject the sub request id header
+		if p.TykAPISpec.EnableContextVars && p.TykAPISpec.GraphQL.ExecutionMode == apidef.GraphQLExecutionModeSupergraph {
+			ctxData := ctxGetData(outreq)
+			if reqID, ok := ctxData["request_id"]; ok {
```
Comment (Aenimus) [inline gateway/reverse_proxy.go:1044]: What do you think about reducing nesting?

`reqID, hasRequestId := ctxData["request_id"]
if !hasRequestId {
    log.Warn("context variables enabled but request_id missing")
} else if requestID, ok := reqID.(string); ok {
    execOptions = append(execOptions, graphql.WithUpstreamHeaders(http.Header{
	 "X-Tyk-Parent-Request-Id": {requestID},
    }))
}`

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4313/merge
Commit: 8504057688c99560ec403b07355ff09c51452825 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3138061308)

**Review by Aenimus — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4313/merge
Commit: 88ad31c636dbff55a37063907be5b608f8d9a767 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3149045507)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4313/merge
Commit: 5bb41c9391495e328ada23a93fb721d0ebd6403b 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3157633601)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4313/merge
Commit: 1febb1603b4c6188ba8258ea81b345c2f891c5a4 
Triggered by: pull_request (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3158977747)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 211d80e695a2db5d80183d68a276abff480f5adb [TT-5852]added sub request id to request (#4313)

* added sub request id to request

* refactored function
Triggered by: push (@kofoworola)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3159119012)

**Pull request merged**